### ISSUE TITLE
Texture Encoder  Network structure

### ISSUE THREAD

**Issue by YQX1996** [facebookresearch/FashionPlus#7]
Hi,
I looked at the network structure of texture encoding. Why need to upsample after downsampling? Will it not lose a lot of details? Is it possible to get the encoding result directly without upsampling? and How is the fashion classifier trained?
Looking forward to your reply.
Thanks.

**Comment by wlhsiao** [comment]
In our experiments, we did not observe any evidence showing lost details due to upsampling.
It is possible to use the encoding without upsampling, but the generator needs to be retrained since it is currently trained to condition on the encoding after upsampling. And using tanh as activation function for the last encoding layer seems to be better than ReLU, so the activation function for the last downsampling layer also needs to be changed. Finally, the segmentation map needs to be downsampled accordingly.

The classifier's network architecture is defined in our supplementary material.
The optimizer we used to train the classifier was adam with initial learning rate 0.001 and weight decay 0.0001. We decay our learning rate with multiplicative factor=0.1 at 60, 80, 100 epoch, and trained the model until epoch 120.

**Comment by YQX1996** [comment]
Hello, 
thank you for your answer!  I find that your colors are all in lab space. Is the classifier lab?
Looking forward to your reply.
Thanks.

**Comment by wlhsiao** [comment]
Right, our cGAN is trained in the Lab color space. For the classifier, it takes in the encodings of clothing pieces as input, not images, so they are not in any color space.

**Comment by YQX1996** [comment]
Hi,
The classifier takes the image coding as the input, and the fashion of clothing in the paper is judged by the classifier. I want to ask how the classifier distinguishes the fashion? Is the input image labeled? 
Looking forward to your reply.
Thanks.

**Comment by wlhsiao** [comment]
The input image is not labeled. Since all images we used are collected from celebrities publicly shared photos or online fashion social platforms, most of them are fashionable and we treat all of them as positive. For more details on training the fashionability classifier, please see Sec.3.2 in our [paper](https://arxiv.org/pdf/1904.09261.pdf).

**Comment by YQX1996** [comment]
Hi,
Have you done experiments in RGB space? Why do you use lab space? I trained the model with RGB, but I couldn't get the experimental results.
Thanks.

**Comment by wlhsiao** [comment]
Yes, we've also done in RGB space, and the original pix2pixHD also trained theirs in RGB. Could you elaborate on what results you couldn't get?

**Comment by YQX1996** [comment]
Hi,
Thank you for your patience, According to the paper, the whole framework includes texture and shape. In structure, shape branches are similar to BicycleGAN. Should I also train BicycleGAN? In addition, I would like to know how big the input image size is when you are training pix2pixhd?

Thanks.


 
---Original---

**Comment by wlhsiao** [comment]
Our shape branch architecture is not that similar as BicycleGAN, since BicycleGAN is also a cGAN, not VAE. You could print out the shape architecture to check. If you want to operate images with different segmentation taxonomy, you would need to re-train the VAE (i.e. shape structure). If you just want to change the texture encoding, you won't need to re-train the VAE.
The image size we train our pix2pixHD with is 256 x 256.
